File: code/11.go
/*
 * @Author: lmio
 * @Date: 2023-02-21 22:22:06
 * @LastEditTime: 2023-04-07 14:34:09
 * @FilePath: /leetcode/code/11.go
 * @Description:11.盛最多的水
 */
package code

import (
	."leetcode/utils"
)

func MaxArea(height []int) int {
	left, right := 0, len(height)-1
	maxarea := 0
	for left < right {
		if height[left] < height[right] {
			maxarea = Max(maxarea, height[left] * (right - left))
			left++
		}else {
			maxarea = Max(maxarea, height[right] * (right - left))
			right--
		}
	}
	return maxarea
}

File: code/11.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-07-27 20:33:29
LastEditors: lmio 2091319361@qq.com
Description: 11. 盛最多水的容器
'''

from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height)-1
        res = 0
        while left < right:
            hl, hr = height[left], height[right]
            if hl < hr:
                res = max(res, (right-left)*hl)
                left += 1
            else:
                res = max(res, (right-left)*hr)
                right -= 1
        return res